modular_skyrat/modules/shotgunrebalance/code/ammobox.dm

```diff
@@ -1,96 +1,154 @@
+	name = "box of flechettes"
+	desc = "A box of 12 gauge flechette shells that specialize in ripping through armor."
 	icon_state = "flechette"
-	ammo_type = /obj/item/ammo_casing/shotgun/flechette
-	max_ammo = 15
 
-/obj/item/ammo_box/advanced/s12gauge/beehive
-	name = "B3-HVE 'Beehive' ammo box"
+/obj/item/storage/box/shotgun/flechette/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/ammo_casing/shotgun/flechette(src)
+
+// BEEHIVE
+/obj/item/storage/box/shotgun/beehive
+	name = "box of B3-HVE 'Beehive' shells"
+	desc = "A box of highly experimental non-lethal shells filled with smart nanite pellets that re-aim themselves when bouncing off from surfaces."
 	icon_state = "beehive"
-	ammo_type = /obj/item/ammo_casing/shotgun/beehive
-	max_ammo = 15
 
-/obj/item/ammo_box/advanced/s12gauge/antitide
-	name = "4NT1-TD3 'Suppressor' ammo box"
+/obj/item/storage/box/shotgun/beehive/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/ammo_casing/shotgun/beehive(src)
+
+// ANTI-TIDE
+/obj/item/storage/box/shotgun/antitide
+	name = "box of 4NT1-TD3 'Suppressor' shells"
+	desc = "A box of highly experimental shells filled with nanite electrodes that will embed themselves in soft targets. \
+			The electrodes are charged from kinetic movement which means moving targets will get punished more."
 	icon_state = "antitide"
-	ammo_type = /obj/item/ammo_casing/shotgun/antitide
-	max_ammo = 15
 
-/obj/item/ammo_box/advanced/s12gauge/iceblox
-	name = "Iceshot ammo box"
+/obj/item/storage/box/shotgun/antitide/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/ammo_casing/shotgun/antitide(src)
+
+// ICEBLOX
+/obj/item/storage/box/shotgun/iceblox
+	name = "box of iceblox shells"
+	desc = "A box of highly experimental shells filled with nanites that will lower the body temperature of hit targets."
 	icon_state = "iceblox"
-	ammo_type = /obj/item/ammo_casing/shotgun/iceblox
-	max_ammo = 15
 
-/obj/item/ammo_box/advanced/s12gauge/incendiary
-	name = "Incendiary Slug ammo box"
+/obj/item/storage/box/shotgun/iceblox/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/ammo_casing/shotgun/iceblox(src)
+
+// INCENDIARY
+/obj/item/storage/box/shotgun/incendiary
+	name = "box of incendiary slugs"
+	desc = "A box of incendiary-coated shotgun slugs."
 	icon_state = "incendiary"
-	ammo_type = /obj/item/ammo_casing/shotgun/incendiary
-	max_ammo = 35
-
-/obj/item/ammo_box/advanced/s12gauge/honk
-	name = "Confetti ammo box"
-	icon_state = "honk"
-	ammo_type = /obj/item/ammo_casing/shotgun/honk
-	max_ammo = 35
+
+/obj/item/storage/box/shotgun/incendiary/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/ammo_casing/shotgun/incendiary(src)
+
+// CONFETTI
+/obj/item/storage/box/shotgun/confetti
+	name = "box of confetti shells"
+	desc = "A box of 12 gauge buckshot shells that have been filled to the brim with confetti."
+	icon_state = "confetti"
+
+/obj/item/storage/box/shotgun/confetti/PopulateContents()
+	for(var/i in 1 to 7)
```

Comment:
Is there a reason why this `for(var/i in 1 to 7)` loop can't be made generic by using a variable that stands for the ammo type instead? Could just make it on the `/obj/item/storage/box/shotgun` and then have the `7` be a variable for how many bullets to spawn.

Reply:
Taking it further, every subtype of `/obj/item/storage/box/shotgun` duplicates basically the same code. There should also be a var on ../shotgun that has the ammo type so you can just do like
```byond
for(var/i in 1 to max_ammo)
	new ammo_type(src)
```
or whatever in the shotgun/PopulateContents()